src/pages/ReportDetailsPage.js

```diff
@@ -56,146 +56,142 @@ const defaultProps = {
     personalDetails: {},
 };
 
-class ReportDetailsPage extends Component {
-    getPolicy() {
-        return this.props.policies[`${ONYXKEYS.COLLECTION.POLICY}${this.props.report.policyID}`];
-    }
+const ReportDetailsPage = (props) => {
+    const policy = useMemo(() => props.policies[`${ONYXKEYS.COLLECTION.POLICY}${props.report.policyID}`], [props.policies, props.report.policyID]);
+    const isPolicyAdmin = PolicyUtils.isPolicyAdmin(policy);
+    const isPolicyExpenseChat = useMemo(() => ReportUtils.isPolicyExpenseChat(props.report), [props.report]);
+    const isChatRoom = useMemo(() => ReportUtils.isChatRoom(props.report), [props.report]);
+    const isThread = useMemo(() => ReportUtils.isThread(props.report), [props.report]);
+    const chatRoomSubtitle = useMemo(() => ReportUtils.getChatRoomSubtitle(props.report), [props.report]);
 
-    getMenuItems() {
-        const menuItems = [
+    const menuItems = useMemo(() => {
+        const items = [
             {
                 key: CONST.REPORT_DETAILS_MENU_ITEM.SHARE_CODE,
                 translationKey: 'common.shareCode',
                 icon: Expensicons.QrCode,
-                action: () => Navigation.navigate(ROUTES.getReportShareCodeRoute(this.props.report.reportID)),
+                action: () => Navigation.navigate(ROUTES.getReportShareCodeRoute(props.report.reportID)),
             },
         ];
 
-        if (ReportUtils.isArchivedRoom(this.props.report)) {
+        if (ReportUtils.isArchivedRoom(props.report)) {
             return [];
         }
 
-        if (lodashGet(this.props.report, 'participants', []).length) {
-            menuItems.push({
+        if (lodashGet(props.report, 'participants', []).length) {
+            items.push({
                 key: CONST.REPORT_DETAILS_MENU_ITEM.MEMBERS,
                 translationKey: 'common.members',
                 icon: Expensicons.Users,
-                subtitle: lodashGet(this.props.report, 'participants', []).length,
+                subtitle: lodashGet(props.report, 'participants', []).length,
                 action: () => {
-                    Navigation.navigate(ROUTES.getReportParticipantsRoute(this.props.report.reportID));
+                    Navigation.navigate(ROUTES.getReportParticipantsRoute(props.report.reportID));
                 },
             });
         }
 
-        if (ReportUtils.isPolicyExpenseChat(this.props.report) || ReportUtils.isChatRoom(this.props.report) || ReportUtils.isThread(this.props.report)) {
-            menuItems.push({
+        if (ReportUtils.isPolicyExpenseChat(props.report) || ReportUtils.isChatRoom(props.report) || ReportUtils.isThread(props.report)) {
```

Comment:
Can we use the variables isPolicyExpenseChat, isChatRoom, and isThread defined at the top in here? Update: looks like we can and we just need to pass it in the list of dependencies to useMemo like we're doing for isThread